utils/sane-backends/Makefile

```diff
@@ -130,6 +130,20 @@ $(call Package/sane-backends/Default)
   USERID=saned:saned
 endef
 
+define Package/sane-daemon-avahi-dbus-service
+$(call Package/sane-backends/Default)
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS+=sane-daemon +avahi-dbus-daemon
+  TITLE+= (network daemon avahi service files)
+endef
+
+define Package/sane-daemon-avahi-nodbus-service
```

Comment:
Sorry for top-posting, but this is a new bit rather than reply:

Perhaps this issue only occurs when using ImageBuilder rather than when 
installing on an existing system?

I noticed the issue with ImageBuilder, not installing on running router.

Regards,

Daniel

On 16-03-28 02:03 PM, Luiz Angelo Daros de Luca wrote:
> In utils/sane-backends/Makefile
> <https://github.com/openwrt/packages/pull/2543#discussion_r57606392>:
>
>> @@ -130,6 +130,20 @@ $(call Package/sane-backends/Default)
>>    USERID=saned:saned
>>  endef
>>
>> +define Package/sane-daemon-avahi-dbus-service
>> +$(call Package/sane-backends/Default)
>> +  SECTION:=utils
>> +  CATEGORY:=Utilities
>> +  DEPENDS+=sane-daemon +avahi-dbus-daemon
>> +  TITLE+= (network daemon avahi service files)
>> +endef
>> +
>> +define Package/sane-daemon-avahi-nodbus-service
>
> I did some tests on PROVIDES:
>
> |
> root@OpenWrt:~# opkg install avahi-daemon
> Installing avahi-dbus-daemon...
> |
>
> So, if no variation defined, it is free to choose any (in this case,
> dbus variation).
>
> After I select one of them (nodbus variant), avahi-daemon is already
> provided
>
> |
> root@OpenWrt:~# opkg install avahi-nodbus-daemon
> (...)
> root@OpenWrt:~# opkg install avahi-daemon
> Package avahi-nodbus-daemon is already installed on root.
> |
>
> I still can ask to install both nodbus and dbus variants, but it will
> fail afterwards because they have conflicting files.
>
> Now using it with DEPENDS:
>
> I installed nut-avahi-service, which depends on avahi-daemon:
>
> With no avahi installed:
>
> opkg install nut-avahi-service
> Installing nut-avahi-service...
> Installing avahi-nodbus-daemon...
> Installing libavahi-nodbus-support...
>
> Now, I installed the dbus variant and the nut-avahi-service after:
>
> |
> opkg install avahi-dbus-daemon nut-avahi-service
> Installing avahi-dbus-daemon
> Installing libavahi-dbus-support
> Installing nut-avahi-service
> |
>
> IMHO, it just works as expected. Why the same thing that works for
> openssh, nut, mpd does not work for sane-backends?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openwrt/packages/pull/2543/files/b6ad5b60004169a2ee30f88022514ceb0f395124..6faee57ddc563cd71dad384dd8c15ea48ab7184c#r57606392>
>